src/TextBox/TextBox.tsx

```diff
@@ -1,39 +1,53 @@
 import * as React from "react";
 import "./text-box-style.scss";
+import { randomId } from "../__utils/randomId";
 
 export interface TextBoxProps {
-    value: string | number;
-    name: string;
+    autoComplete?: "on" | "off";
+    className?: string;
+    disabled?: boolean;
+    error?: string;
+    focus?: boolean;
     id?: string;
-    pattern?: string;
-    required?: boolean;
-    minLength?: number;
+    label?: string;
     maxLength?: number;
+    minLength?: number;
+    name: string;
+    onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
     onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
     onKeyDown?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
-    onKeyUp?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
     onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
-    onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
-    onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
-    type?: string;
-    label?: string;
-    error?: string;
-    placeHolder?: string;
-    className?: string;
-    focus?: boolean;
-    readonly?: boolean;
-    disabled?: boolean;
-    autoComplete?: boolean;
+    onKeyUp?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
+    pattern?: string;
+    placeholder?: string;
+    readOnly?: boolean;
     reference?: React.RefObject<HTMLInputElement>;
+    required?: boolean;
+    success?: boolean;
+    type?: string;
+    value: string | number;
+    showErrorMessage?: boolean;
 }
 
 export const TextBox: React.FunctionComponent<TextBoxProps> = (props: TextBoxProps): React.ReactElement<void> => {
+    const [id, setId] = React.useState<string>(null);
+    const [showErrorMessage, setShowErrorMessage] = React.useState<boolean>(true);
+
+    React.useEffect(() => {
+        setId(props.id ? props.id : (props.label ? randomId("tbg-") : null));
+    }, [props.id, props.label]);
+
+    React.useEffect(() => {
+        setShowErrorMessage(props.showErrorMessage === undefined || props.showErrorMessage === null ? true : !!props.showErrorMessage);
```

Comment:
```suggestion
        setShowErrorMessage(!!props.showErrorMessage);
```

Reply:
The purpose of this logic is to set a default of `true` if `showErrorMessage` is not passed (`undefined` or `null`). So your suggestion will set it to `false` if the user didn't pass this property which is not the expected behavior